### ISSUE TITLE
Issues with sequence to sequence models with Longformer as encoder and decoder 

### ISSUE THREAD

**Issue by Hellisotherpeople** [ThilinaRajapakse/simpletransformers#596]
**Describe the bug**
1.For sequence to sequence models done without a unified encoder decoder, it appears that the current code specifies using the BertTokenizer no matter what for decoding. I had to edit that part of the code to use the LongFormertokenizer. Maybe I should submit a PR to properly use the right tokenizer for any decoder? 
2. Using the following code snippet, I am getting a bug after one pass that it cannot find my cached model (because it appears to be failing to save it). Does anyone know what the issue is? Any work around? setting use_cache = False didn't work... 

**To Reproduce**
Run this code (get your own sequence to sequence dataset): 

```
import pandas as pd
import logging
from sklearn.model_selection import train_test_split
from simpletransformers.seq2seq import Seq2SeqModel
logging.basicConfig(level=logging.ERROR)
transformers_logger = logging.getLogger("transformers")
transformers_logger.setLevel(logging.ERROR)


model_args = {
    "reprocess_input_data": True,
    "overwrite_output_dir": True,
    "max_seq_length": 1000,
    "train_batch_size": 1,
    "num_train_epochs": 10,
    "save_eval_checkpoints": False,
    "save_model_every_epoch": True,
    "evaluate_during_training": True,
    "evaluate_generated_text": True,
    "evaluate_during_training_verbose": True,
    "use_multiprocessing": False,
    "max_length": 300,
    "manual_seed": 4,
    "fp16" : False,
    "cache_dir" : "/home/lain/lain/CX_DB8/cache_dir"

}

model = Seq2SeqModel(
    encoder_type="longformer",
    encoder_name= "allenai/longformer-base-4096",
    decoder_name = "allenai/longformer-base-4096",
    args=model_args,
)

my_debate_data = pd.read_csv("/home/lain/lain/CX_DB8/card_data/debate2019.csv")[['Full-Document', "Extract"]]
my_debate_data.columns = ["input_text", "target_text"]
train, test = train_test_split(my_debate_data, test_size=0.2)
model.train_model(train, eval_data = test)

results = model.eval_model(test)
```

**Expected behavior**
Sequence to sequence model trains, saves and runs properly

**Desktop (please complete the following information):**
Ubuntu 20.04 

**Additional context**

**Comment by ThilinaRajapakse** [comment]
1. I think this was probably missed out when Longformer was added. A PR would be great!

2. It may have something to do with using the full path here -- `"cache_dir" : "/home/lain/lain/CX_DB8/cache_dir"`. Can you try using a relative path?

**Comment by Hellisotherpeople** [comment]
Let me try again with a different cache directory, and I'll try to get a PR for resolving this soon! I'll keep this issue open for now. Thank you so much for your support.

**Comment by davidlenz** [comment]
Was able to circumvent the problem that cache_data could not be written to disk by creating the folder beforehand in [line 63](https://github.com/ThilinaRajapakse/simpletransformers/blob/d405b4aba23175c81a2e12db60c2dfd7a60fd9b4/simpletransformers/seq2seq/seq2seq_utils.py#L63) using `os.makedirs(os.path.dirname(os.path.abspath(cached_features_file)), exist_ok=True)`

However then the problem reported in [this issue on huggingfance/transformers](https://github.com/huggingface/transformers/issues/5809) came up. Seems to be the case that longformer doesn't work in encoder decoder setting yet.
